assets/app/features/upgrade/controllers/upgrade7-openstack-services.controller.js

```diff
@@ -0,0 +1,100 @@
+(function() {
+    'use strict';
+
+    /**
+    * @ngdoc function
+    * @name crowbarApp.controller:Upgrade7OpenStackServicesController
+    * @description
+    * # Upgrade7OpenStackServicesController
+    * This is the controller used on the Stop OpenStack Services page
+    */
+    angular.module('crowbarApp')
+        .controller('Upgrade7OpenStackServicesController', Upgrade7OpenStackServicesController);
+
+    Upgrade7OpenStackServicesController.$inject = ['$translate', 'openStackFactory'];
+    // @ngInject
+    function Upgrade7OpenStackServicesController($translate, openStackFactory) {
+        var vm = this;
+            
+
+        vm.openStackServices = {
+            completed: false,
+            valid: false,
+            stopServicesErrorMessage: false,
+            checks: {
+                services: {
+                    status: false, 
+                    label: 'upgrade7.steps.openstack-services.codes.services'
+                },
+                backup: {
+                    status: false, 
+                    label: 'upgrade7.steps.openstack-services.codes.backup'
+                }
+            },
+            runOpenStackServices: runOpenStackServices
+        };
+
+
+        /**
+         *  Validate OpenStackServices required for Cloud 7 Upgrade
+         */
+        function runOpenStackServices() {
+   
+            openStackFactory.stopOpenstackServices()
+                .then(
+                    //Success handler. Stop all OpenStackServices successfully:
+                    function(openStackServicesResponse) {
```

Comment:
This is getting too complicated. Could please define stopOpenStack and createOpenStachBackup handler functions separately?